### PR TITLE
Add Redux in the project.

### PR DESCRIPTION
**Pull request by addisalemseifu**
-Used [ducks pattern](https://github.com/erikras/ducks-modular-redux) file structure style.
-Added `states`, `actions`, and `reducer` functions for the `bookSlice.js` file and `categoriesSlice.js` file.
-Update the `Readme. md` file.

### CODE REVIEW

File: src/redux/books/bookSlice.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      };
+    case BOOK_REMOVED:
+      return {
+        books: state.pop(),
+      };
+    default:
+      return state;
+  }
+};
+
```
Comment (Ibizugbe) [inline src/redux/books/bookSlice.js]: Kindly use the `createSlice` method from `@reduxjs/toolkit`  to add your reducers so as to have a single or combined reducer that can be exported👍. This function helps to reduce the boilerplate required to add data to redux in a canonical way😉

you can use this method by importing it from `@reduxjs/toolkit`  like this `import { createSlice } from "@reduxjs/toolkit"` and kindly export the various actions and reducer👍.

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      return {
+        state,
+      };
+    default:
+      return state;
+  }
+};
+
+categoriStatus();
+reducer();
```
Comment (Ibizugbe) [inline src/redux/categories/categoriesSlice.js]: Just as I requested in the previous review, Kindly use the createSlice method from @reduxjs/toolkit to add your reducers so as to have a single or combined reducer that can be exported👍. This function helps to reduce the boilerplate required to add data to redux in a canonical way😉

you can use this method by importing it from @reduxjs/toolkit like this import { createSlice } from "@reduxjs/toolkit" and kindly export the various actions and your reducer👍.

Comment (Ibizugbe) [inline src/redux/categories/categoriesSlice.js]: Just as I requested in the previous review, Kindly use the createSlice method from @reduxjs/toolkit to add your reducers so as to have a single or combined reducer that can be exported👍. This function helps to reduce the boilerplate required to add data to redux in a canonical way😉

you can use this method by importing it from @reduxjs/toolkit like this import { createSlice } from "@reduxjs/toolkit" and kindly export the various actions and your reducer👍.

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @addisalemseifu 👋👋👋, 

tryagain


Good job so far!👍
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Linters are properly setup and they are passing👍.
- [x] `Git flow` have been used👍.
- [x] `@reduxjs/toolkit` & `react-redux` have been properly setup👍.

## Required Changes ♻️

- [ ] Just as required in this milestone of the project, kindly structure your application files using a "feature folder" approach and use the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for your Redux files👍.
image

- [ ] Just as required in this milestone, kindly update your  [src/redux/store.js](https://github.com/addisalemseifu/react-bookstore/pull/2/files#diff-e716f17be314ea12f823ad2fa506050ffc72214c90694821dcedc7eaa19f20e8) file to have a combined reducer for your books and categories👍.
image


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @addisalemseifu,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**